src/main/java/cs2103/v15_1j/jimjim/command/DeleteCommand.java

```diff
@@ -28,7 +28,12 @@ public char getPrefix() {
     @Override
     public UIFeedback undo(ControllerStates conStates) {
 		// Add task/event back at former position
-	    conStates.masterList.add(taskNum-1, backup);
+	    conStates.masterList.add(backup);
+	    
+	    if(!conStates.searchResultsList.isEmpty()){
```

Comment:
@SorataYuu shouldn't this only happen if the item is on search results originally? And even if that's the case, what if i have changed my search criteria such that the item no longer passes all the filters?